inspectit-oce-agent/src/main/java/rocks/inspectit/oce/agent/AgentMain.java

```diff
@@ -0,0 +1,113 @@
+package rocks.inspectit.oce.agent;
+
+import rocks.inspectit.oce.bootstrap.AgentManager;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.instrument.Instrumentation;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.security.AllPermission;
+import java.security.CodeSource;
+import java.security.PermissionCollection;
+import java.util.jar.JarFile;
+
+/**
+ * Entry point of the agent.
+ *
+ * @author Jonas Kunz
+ */
+public class AgentMain {
+
+    private static final String INSPECTIT_BOOTSTRAP_JAR_PATH = "/inspectit-oce-bootstrap.jar";
+    private static final String INSPECTIT_CORE_JAR_PATH = "/inspectit-oce-core.jar";
+
+    public static void agentmain(String agentArgs, Instrumentation inst) {
+        //TODO: currently replacing the agent does not really work as all Agent versions share the same namespace in the same classpath
+        premain(agentArgs, inst);
+    }
+
+    public static void premain(String agentArgs, Instrumentation inst) {
+        try {
+            ClassLoader icl = initializeClasspath(inst);
+            AgentManager.startOrReplaceInspectitCore(icl);
```

Comment:
this still works right, we are able to reference anything from the bootstrap here or? I was thinking that this `AgentManager` would be loaded by the class loader loading the `AgentMain`.. But this seems not to be the case, I tested it.. but actually, this is also how it was in the old one..

Reply:
This also represents a problem if we would like to push all dependencies to our class loader. Maybe the easiest is to load the `AgentManager` using reflection from the `icl`, and also invoke this method using reflection? Then you could also remove the dependency to the `inspectit-oce-bootstrap` in this project and we keep it totally separated..